tensorboard/backend/application.py

```diff
@@ -265,7 +266,32 @@ def __init__(self, plugins, path_prefix=''):
         else:
           path = (self._path_prefix + DATA_PREFIX + PLUGIN_PREFIX + '/' +
                   plugin.plugin_name + route)
-        self.data_applications[path] = app
+
+        if path.endswith('/*'):
+          # Note we remove the '*' but leave the slash in place.
+          path = path[:-1]
+          if '*' in path:
+            # note we re-add the removed * in the format string
+            raise ValueError('Plugin %r handles invalid route %r*:  Only '
```

Comment:
nit: Parallel errors in this file (and generally throughout TensorBoard)
use one space after a period.

Reply:
done